src/sage/matrix/matrix_laurent_mpolynomial_dense.pyx

```diff
@@ -0,0 +1,125 @@
+"""
+Dense matrices over multivariate polynomials over fields.
+
+AUTHOR:
+
+- Enrique Artal (2023-??): initial version
+"""
+
+# *****************************************************************************
+#       Copyright (C) 2023 Enrique Artal <artal@unizar.es>
+#
+#  Distributed under the terms of the GNU General Public License (GPL)
+#  as published by the Free Software Foundation; either version 2 of
+#  the License, or (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# *****************************************************************************
+from sage.matrix.matrix_generic_dense cimport Matrix_generic_dense
+
+# from sage.matrix.matrix2 cimport Matrix
+#
+# from sage.matrix.constructor import identity_matrix
+# from sage.rings.polynomial.laurent_polynomial_ring_base import LaurentPolynomialRing_generic
+# from sage.rings.polynomial.multi_polynomial_libsingular cimport MPolynomialRing_libsingular
+# from sage.rings.polynomial.polynomial_singular_interface import can_convert_to_singular
+#
+# from sage.libs.singular.function import singular_function, lib
+#
+# from cysignals.signals cimport sig_on, sig_off
+
+from sage.matrix.constructor import identity_matrix
+from sage.rings.polynomial.laurent_polynomial_ring_base import LaurentPolynomialRing_generic
+
+cdef class Matrix_laurent_mpolynomial_dense(Matrix_generic_dense):
```

Comment:
This should have a corresponding definition in a `matrix_laurent_mpolynomial_dense.pxd` file.

Reply:
Done, just copying and editing matrix_mpolynomial_dense.pxd